opensearch_py_ml/ml_commons/rag_pipeline/rag/AIConnectorHelper.py

```diff
@@ -0,0 +1,698 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+
+import boto3
+from botocore.exceptions import BotoCoreError
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+from requests_aws4auth import AWS4Auth
+import time
+from opensearchpy import OpenSearch, RequestsHttpConnection
+
+# This Python code is compatible with AWS OpenSearch versions 2.9 and higher.
+class AIConnectorHelper:
+    def __init__(self, region, opensearch_domain_name, opensearch_domain_username,
+                 opensearch_domain_password, aws_user_name, aws_role_name):
+        """
+        Initialize the AIConnectorHelper with necessary AWS and OpenSearch configurations.
+        """
+        self.region = region
+        self.opensearch_domain_name = opensearch_domain_name
+        self.opensearch_domain_username = opensearch_domain_username
+        self.opensearch_domain_password = opensearch_domain_password
+        self.aws_user_name = aws_user_name
+        self.aws_role_name = aws_role_name
+
+        # Retrieve the OpenSearch domain endpoint and ARN
+        domain_endpoint, domain_arn = self.get_opensearch_domain_info(self.region, self.opensearch_domain_name)
+        if domain_arn:
+            self.opensearch_domain_arn = domain_arn
+        else:
+            print("Warning: Could not retrieve OpenSearch domain ARN.")
+            self.opensearch_domain_arn = None
+
+        if domain_endpoint:
+            # Construct the full domain URL
+            self.opensearch_domain_url = f'https://{domain_endpoint}'
```

Comment:
better to get the total domain url I think. What if somebody wants to use `http`?